File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/queryhandler/EnrollmentQueryController.java
package cloud.ambar.creditcard.enrollment.queryhandler;

import cloud.ambar.common.queryhandler.QueryController;
import cloud.ambar.common.projection.MongoTransactionalProjectionOperator;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.context.annotation.RequestScope;

@RestController
@RequestScope
@RequestMapping("/api/v1/credit_card/enrollment")
public class EnrollmentQueryController extends QueryController {
    private final GetUserEnrollmentsQueryHandler getUserEnrollmentsQueryHandler;

    public EnrollmentQueryController(
            MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator,
            GetUserEnrollmentsQueryHandler getUserEnrollmentsQueryHandler
    ) {
        super(mongoTransactionalProjectionOperator);
        this.getUserEnrollmentsQueryHandler = getUserEnrollmentsQueryHandler;
    }

    @PostMapping(value = "/list-enrollments")
    public Object listEnrollments(
            @RequestHeader("X-With-Session-Token") String sessionToken
    ) {
        GetUserEnrollmentsQuery query = GetUserEnrollmentsQuery.builder()
                .sessionToken(sessionToken)
                .build();

        return processQuery(query, getUserEnrollmentsQueryHandler);
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Query/EnrollmentQueryController.cs
using CreditCardEnrollment.Common.Projection;
using CreditCardEnrollment.Common.Query;
using Microsoft.AspNetCore.Mvc;

namespace CreditCardEnrollment.CreditCard.Enrollment.Query;

[ApiController]
[Route("api/v1/credit_card/enrollment")]
[Produces("application/json")]
[Consumes("application/json")] 
public class EnrollmentQueryController : QueryController {
    private readonly GetUserEnrollmentsQueryHandler _getUserEnrollmentsQueryHandler;

    public EnrollmentQueryController(
        MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator,
        ILogger<EnrollmentQueryController> logger,
        GetUserEnrollmentsQueryHandler getUserEnrollmentsQueryHandler
    ) : base(mongoTransactionalProjectionOperator, logger) {
        _getUserEnrollmentsQueryHandler = getUserEnrollmentsQueryHandler;
    }

    [HttpPost("list-enrollments")]
    [ProducesResponseType(typeof(object), StatusCodes.Status200OK)]
    public IActionResult ListEnrollments(
        [FromHeader(Name = "X-With-Session-Token")] string sessionToken) {
        var query = new GetUserEnrollmentsQuery {
            SessionToken = sessionToken
        };

        return new OkObjectResult(ProcessQuery(query, _getUserEnrollmentsQueryHandler));
    }
}